source/slang/slang-check-overload.cpp

```diff
@@ -1590,10 +1584,17 @@ namespace Slang
             //          float g(float y) { return f(y); }   // will call S::f() instead of ::f()
             //      }
             //  we will count the distance from the reference site to the declaration in the scope tree.
-
-            auto scopeRank = getScopeRank(left->item.declRef, right->item.declRef, this->m_outerScope);
-            if (scopeRank)
-                return scopeRank;
+            //  NOTE: DON'T do this for the generic function, because for the generic function compare, we only check the
+            //  template parameter/arguments, not the function parameters/arguments. This could break the assumption
+            //  for this logic where we assume that all the candidates are valid.
+            //  This logic will be evaluated later when "Flavor" becomes "Func" or "Expr", because by then the type checks for
+            //  the candidate will be done.
+            if (left->flavor != OverloadCandidate::Flavor::Generic && left->flavor != OverloadCandidate::Flavor::UnspecializedGeneric)
```

Comment:
what if right is generic?

Reply:
Hmm, that is my oversight. I think the left and right should always the same flavor. Isn't that the case?

Reply:
updated.